src/main/java/org/opensrp/web/rest/ImportBulkDataResource.java

```diff
@@ -0,0 +1,116 @@
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Controller
+@RequestMapping("/rest/import")
+public class ImportBulkDataResource {
+
+	private ImportBulkDataService importBulkDataService;
+
+	private static final String SAMPLE_CSV_FILE = "./importsummaryreport.csv";
+
+	private static Logger logger = LoggerFactory.getLogger(ImportBulkDataResource.class.toString());
+
+	@Autowired
+	public void setImportBulkDataService(ImportBulkDataService importBulkDataService) {
+		this.importBulkDataService = importBulkDataService;
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE }, value = "/organizations")
+	public ResponseEntity importOrganizationsData(@RequestParam("file") MultipartFile file)
+			throws IOException {
+
+		List<Map<String, String>> csvClients = readCSVFile(file);
+		CsvBulkImportDataSummary csvBulkImportDataSummary = importBulkDataService
+				.convertandPersistOrganizationdata(csvClients);
+
+		generateCSV(csvBulkImportDataSummary);
+		File csvFile = new File(SAMPLE_CSV_FILE);
+
+		return ResponseEntity.ok()
+				.header("Content-Disposition", "attachment; filename=" + "importsummaryreport" + ".csv")
+				.contentLength(csvFile.length())
+				.contentType(MediaType.parseMediaType("text/csv"))
+				.body(new FileSystemResource(csvFile));
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE }, value = "/practitioners")
+	public ResponseEntity importPractitionersData(@RequestParam("file") MultipartFile file) throws
+			IOException {
+
+		List<Map<String, String>> csvClients = readCSVFile(file);
+		CsvBulkImportDataSummary csvBulkImportDataSummary = importBulkDataService
+				.convertandPersistPractitionerdata(csvClients);
+
+		generateCSV(csvBulkImportDataSummary);
+		File csvFile = new File(SAMPLE_CSV_FILE);
+
+		return ResponseEntity.ok()
+				.header("Content-Disposition", "attachment; filename=" + "importsummaryreport" + ".csv")
+				.contentLength(csvFile.length())
+				.contentType(MediaType.parseMediaType("text/csv"))
+				.body(new FileSystemResource(csvFile));
+	}
+
+	private List<Map<String, String>> readCSVFile(MultipartFile file) throws IOException {
+		List<Map<String, String>> csvClients = new ArrayList<>();
+		try (Reader reader = new InputStreamReader(file.getInputStream());
+		     CSVParser parser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader())) {
+
+			List<CSVRecord> records = parser.getRecords();
+			for (CSVRecord record : records) {
+				csvClients.add(record.toMap());
+			}
+		}
+		return csvClients;
+	}
+
+	private void generateCSV(CsvBulkImportDataSummary csvBulkImportDataSummary) {
+		try (
+				BufferedWriter writer = Files.newBufferedWriter(Paths.get(SAMPLE_CSV_FILE));
+				CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT);
```

Comment:
If 2 users are importing files, I think this will lead to importing returning the same file. I think you can use the filename with and concatenate with timestamp to reduce chances of conflicts

Reply:
Done